### PR TITLE
Merge build workflows and update badges

### PR DESCRIPTION
**Pull request by qweeah**
Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,16 +4,23 @@ on:
...
+      fail-fast: true
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: setup go environment
+      - name: Setup Go ${{ matrix.go-version }}
         uses: actions/setup-go@v3
         with:
-          go-version: '1.18.1'
+          go-version: ${{ matrix.go-version }}
```
Comment (shizhMSFT) [inline .github/workflows/build.yml:23]: Are you using the latest go version of `1.18`?

**Review by shizhMSFT — COMMENTED** [review]
Have you updated the badge link?

**Comment by qweeah** [comment]
> Have you updated the badge link?

Added

File: .github/workflows/build.yml (hunk 1 above)
Comment (qweeah) [reply .github/workflows/build.yml:23]: I am using 1.18.1 locally but the build was based on 1.18.2. Is there a concern?

Comment (shizhMSFT) [reply .github/workflows/build.yml:23]: Could you checkout the `check-latest` option for `actions/setup-go@v3`?

Comment (qweeah) [reply .github/workflows/build.yml:23]: Added

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**